_source/_posts/2022-08-12-ci-cd-circleci-spinnaker-microservices.md

```diff
@@ -0,0 +1,949 @@
+        'pods/log',
+      ]
+    verbs: ['get', 'list']
+  - apiGroups: ['']
+    resources: ['pods', 'services', 'secrets', 'configmaps', 'persistentvolumeclaims']
+    verbs:
+      [
+        'create',
+        'delete',
+        'deletecollection',
+        'get',
+        'list',
+        'patch',
+        'update',
+        'watch',
+      ]
+  - apiGroups: ['autoscaling']
+    resources: ['horizontalpodautoscalers']
+    verbs: ['list', 'get']
+  - apiGroups: ['apps']
+    resources: ['controllerrevisions']
+    verbs: ['list']
+  - apiGroups: ['extensions', 'apps']
+    resources: ['daemonsets', 'deployments', 'deployments/scale', 'ingresses', 'replicasets', 'statefulsets']
+    verbs:
+      [
+        'create',
+        'delete',
+        'deletecollection',
+        'get',
+        'list',
+        'patch',
+        'update',
+        'watch',
+      ]
+  # These permissions are necessary for Halyard to operate. We use this role also to deploy Spinnaker itself.
+  - apiGroups: ['']
+    resources: ['services/proxy', 'pods/portforward']
+    verbs:
+      [
+        'create',
+        'delete',
+        'deletecollection',
+        'get',
+        'list',
+        'patch',
+        'update',
+        'watch',
+      ]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jhipster-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: jhipster-role
+subjects:
+  - namespace: demo
+    kind: ServiceAccount
+    name: jhipster-service-account
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: jhipster-service-account
+  namespace: demo
+```
+
+Notice there are some differences with the Spinnaker service account. The k8s manifests generated by JHipster include the creation of `secrets`, `configmaps`, and `persistentvolumeclaims`, and the management of those Kubernetes resources must be granted to the service account used for deploying the JHipster k8s manifests.
+
+Create the Kubernetes `jhipster-service-account`:
+
+```shell
+CONTEXT=$(kubectl config current-context)
+
+kubectl apply --context $CONTEXT \
+    -f ./jhipster-service-account.yml
+TOKEN=$(kubectl get secret --context $CONTEXT \
+   $(kubectl get serviceaccount jhipster-service-account \
+       --context $CONTEXT \
+       -n demo \
+       -o jsonpath='{.secrets[0].name}') \
+   -n demo \
+   -o jsonpath='{.data.token}' | base64 --decode)
+
+kubectl config set-credentials ${CONTEXT}-token-user --token $TOKEN
+
+kubectl config set-context $CONTEXT --user ${CONTEXT}-token-user
+```
+
+Add the account to the Halyard configuration:
+
+```shell
+CONTEXT=$(kubectl config current-context)
+
+hal config provider kubernetes account add jhipster-gke-account \
+    --context $CONTEXT
+```
+
+### Add a Docker registry account
+
+As the goal is to trigger the pipeline execution when a new image is pushed to Docker Hub, you need to configure a docker-registry provider with Halyard.
```

Comment:
```suggestion
As the goal is to trigger the pipeline execution when a new image is pushed to Docker Hub, you need to configure a Docker-registry provider with Halyard.
```

Reply:
In this case, the term is generic. A docker-registry provider can be Docker or Google.

Reply:
I googled for the term "docker registry" and it appears to be a product term so should be spelled Docker Registry. hal does have a command called `docker-registry`, so that might be where the confusion comes in. I'm going commit the change from "docker-registry" to "Docker Registry" since this is not a command in this sentence.